nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/rs2d/RS2DData.java

```diff
@@ -235,29 +240,27 @@ private void openParFile(String parpath, boolean processed) throws IOException {
 
         Path headerPath = Paths.get(parpath, HEADER_FILE_NAME);
         Path seriesPath = Paths.get(parpath, SERIES_FILE_NAME);
-        try {
-            header = new RS2DHeader(headerPath);
+        try (InputStream input = Files.newInputStream(headerPath)) {
+            header = new HeaderParser().parse(input);
             if (seriesPath.toFile().exists()) {
                 seriesDocument = readDocument(seriesPath);
             }
             groupDelay = readGroupDelay();
-            obsNucleus = header.getString(OBSERVED_NUCLEUS);
+            obsNucleus = header.get(OBSERVED_NUCLEUS).stringValue();
             Double obsFreq = null;
-            Double obsSW = header.getDouble(SPECTRAL_WIDTH);
-            tempK = header.getDouble(SAMPLE_TEMPERATURE);
-            String baseSFName = "";
+            double obsSW = header.<NumberValue>get(SPECTRAL_WIDTH).getValueAsHertz(header);
```

Comment:
This is the reason for the change: Spinlab data files can store SW as Hz or Ppm, reading the value directly from the xml is error-prone.